File: my_py_pkg/my_py_pkg/counter_reset.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import SetBool
from functools import partial


class CounterResetClientNode(Node):
    def __init__(self):
        super().__init__("counter_reset")
        self.call_reset_counter_server(True)
        self.call_reset_counter_server(True)
        self.call_reset_counter_server(False)

    def call_reset_counter_server(self, data):
        client = self.create_client(SetBool, "reset_counter")

        while not client.wait_for_service(1.0):
            self.get_logger().warn("Waiting for the server to be up...")

        request = SetBool.Request()
        request.data = data

        future = client.call_async(request)
        future.add_done_callback(
            partial(self.callback_reset_counter, data=data))

    def callback_reset_counter(self, future, data):
        try:
            response = future.result()
            self.get_logger().info(str(response.message))
        except Exception as e:
            self.get_logger().error("Service call failed %r" % (e,))


def main(args=None):
    rclpy.init(args=args)
    node = CounterResetClientNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: my_cpp_pkg/src/counter_reset.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/set_bool.hpp"

class CounterResetClientNode : public rclcpp::Node
{
public:
    CounterResetClientNode() : Node("counter_reset")
    {
        threads_.push_back(std::thread(std::bind(&CounterResetClientNode::callCounterResetService, this, true)));
        threads_.push_back(std::thread(std::bind(&CounterResetClientNode::callCounterResetService, this, false)));
        threads_.push_back(std::thread(std::bind(&CounterResetClientNode::callCounterResetService, this, true)));
        threads_.push_back(std::thread(std::bind(&CounterResetClientNode::callCounterResetService, this, true)));
        threads_.push_back(std::thread(std::bind(&CounterResetClientNode::callCounterResetService, this, false)));
    }

    void callCounterResetService(bool data)
    {
        auto client = this->create_client<example_interfaces::srv::SetBool>("reset_counter");
        while (!client->wait_for_service(std::chrono::seconds(1)))
        {
            RCLCPP_WARN(this->get_logger(), "Waitung for the server to be up...");
        }

        auto request = std::make_shared<example_interfaces::srv::SetBool::Request>();
        request->data = data;

        auto future = client->async_send_request(request);

        try
        {
            auto response = future.get();
            RCLCPP_INFO(this->get_logger(), "Response: %s", response->message.c_str());
        }
        catch (const std::exception &e)
        {
            RCLCPP_ERROR(this->get_logger(), "Service call failed.");
        }
    }

private:
    std::vector<std::thread> threads_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<CounterResetClientNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}